contracts/samples/AuthorizedAddresses.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.7;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+abstract contract AuthorizedAddresses is Ownable{
+
+    address[] public authorizedAddresses;
+
+
+    modifier onlyAuthorizedAddresses() {
+        bool isAuthorized = false;
+        for(uint i = 0; i < authorizedAddresses.length; i++) {
+            if(authorizedAddresses[i] == msg.sender) {
+                isAuthorized = true;
+            }
+        }
+        require(isAuthorized);
+        _;
+    }
+
+    function addAuthorizedAddress(address addr) public onlyOwner{
+        authorizedAddresses.push(addr);
```

Comment:
Emit an event when an address is added to the list of authorized addresses:
```
emit AuthorizationGranted(_addr);
```

You'll need to define the event at the top of the contract:
```
/**
 * @notice Event triggered when an account is added to the authorized addresses list.
 * @param addr the address of the account to add to the list.
 */
event AuthorizationGranted(address indexed addr);
```